mono/mini/exceptions-arm.c

```diff
@@ -14,9 +14,9 @@
 #include <string.h>
 
 #ifndef MONO_CROSS_COMPILE
-#ifdef HAVE_ASM_SIGCONTEXT_H
+#ifdef PLATFORM_ANDROID
 #include <asm/sigcontext.h>
-#endif  /* def HAVE_ASM_SIGCONTEXT_H */
+#endif  /* def PLATFORM_ANDROID */
 #endif
 
```

Comment:
Yes, the purpose is to avoid including this header file, Alpine sets HAVE_ASM_SIGCONTEXT_H and including it causes struct redefinition errors as the types are already defined in signal.h.

From the commit message:
```
[mini] Fix asm/sigcontext.h include in exceptions-arm.h

We include the header under the PLATFORM_ANDROID ifdef in mono-sigcontext.h as well [1].
It was probably added because older Bionic didn't include this from signal.h [2].

Fixes the build on Alpine/musl-libc.

[1] https://github.com/mono/mono/blob/19a99abf1680c85f8ff58c2b44ea8c15bb557ca5/mono/utils/mono-sigcontext.h#L5
[2] https://bugzilla.mozilla.org/show_bug.cgi?id=859580
```